### PR TITLE
Don't stop retrieving data from Dependency-Track after the first page.

### LINKED ISSUE #9751: Quality-time does not retrieve more than one page from Dependency Track
Tasks:
- [x] Fix pagination code
- [x] Fix async loop
- [x] Add a warning if the configured project does not exist (e.g. typo)

### PR DESCRIPTION
**Pull request by fniessink**
Fixes #9751.

### CODE REVIEW

File: components/collector/src/source_collectors/dependency_track/base.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ async def _get_project_uuids(self) -> dict[str, str]:
 
     async def _get_projects(self) -> AsyncIterator[DependencyTrackProject]:
         """Return the Dependency-Track projects."""
-        projects_api = URL(await self._api_url() + "/project")
+        projects_api = URL(await DependencyTrackBase._api_url(self) + "/project")
```
Comment (wkoot) [inline components/collector/src/source_collectors/dependency_track/base.py:62]: Why was this needed?

File: components/collector/src/source_collectors/dependency_track/base.py [hunk 2]
```diff
@@ -44,8 +44,8 @@ async def _get_source_responses(self, *urls: URL) -> SourceResponses:
         responses = SourceResponses()
         for url in urls:
             page_nr = 1  # Page numbers start at 1
-            total_count = self.PAGE_SIZE + 1  # Total count is still unknown, make it so big we get at least one page
-            while page_nr * self.PAGE_SIZE < total_count:
+            total_count = 1  # Total count is still unknown, make sure we retrieve at least one page
+            while (page_nr - 1) * self.PAGE_SIZE < total_count:
                 offsetted_url = URL(f"{url}{"&" if "?" in url else "?"}pageSize={self.PAGE_SIZE}&pageNumber={page_nr}")
```
Comment (wkoot) [inline components/collector/src/source_collectors/dependency_track/base.py]: This url string munging is a bit dirty, we should have a helper method to merge http GET params from the query.
There are a lot of helpers provided by `urllib.parse`, avoiding to worry about a `#` fragment in the url.

File: components/collector/src/source_collectors/dependency_track/base.py (hunk 1 above)
Comment (fniessink) [reply components/collector/src/source_collectors/dependency_track/base.py:62]: Because the code was wrong :-) 

If a subclass would override the `_api_url` method to, for example, access the `/issues` endpoint, this method would also call that overridden method and try to access `/issues/project`. 

It was just a coincidence we haven't triggered this accident waiting to happen so far: only `DependencyTrackSourceUpToDateness` overrides `_api_url` at the moment, but it doesn't call `_get_projects`. The other subclasses do call `_get_projects` but don't override `_api_url`. Pfew!

File: components/collector/src/source_collectors/dependency_track/base.py (hunk 2 above)
Comment (fniessink) [reply components/collector/src/source_collectors/dependency_track/base.py]: I don't see a helper in `urllib.parse` that lets one add one query argument to a URL. But I agree a helper is useful. I'll add one.
